### ISSUE TITLE
MacOS support

### ISSUE THREAD

**Issue by vbauer** [chunkypixel/batariBasic-extension#1]
I've just checked version 0.1.0 on MacOS and faced with this problem:

Run command "bB: Compile batari Basic source code":
```
Building file 'ex_dpc_template.bas'...
/bin/sh: /Users/vbauer/.vscode/extensions/chunkypixel.bataribasic-0.1.0/outbinBB.1.1d.reveng41/2600bas.bat: No such file or directory
Process exit code: 127
```

Unfortunately, I don't have folder "outbinBB.1.1d.reveng41":
```bash
➜  chunkypixel.bataribasic-0.1.0 pwd
/Users/vbauer/.vscode/extensions/chunkypixel.bataribasic-0.1.0
➜  chunkypixel.bataribasic-0.1.0 ls
CHANGELOG.md LICENSE.txt  NOTES        README.md    icon.pdn     images       node_modules out          package.json snippets     syntaxes
``` 

It also seems to me that startup script `2600bas.bat` should be `2600bas.sh` or something like that

Tha same problem with `dasm`:
```
Building file 'pong.asm'...
/bin/sh: /Users/vbauer/.vscode/extensions/chunkypixel.bataribasic-0.1.0/outbinDasm/dasm.exe: No such file or directory
Process exit code: 127
```

**PS:**
I know that you haven't announced MacOS support, but probably you'll take a look at this issue 
Thank you for this plugin, I think it's a great project and it will have future!

**Comment by chunkypixel** [comment]
Hi,
Thanks for the info. Yes definitely want to have macOS and Linux as part of the extension. I need to install Linux to start working on an updated release and will do so when I have an opportunity - I believe macOS should work once this is complete.

**Comment by chunkypixel** [comment]
Hi vbauer,

Please try the latest build (v0.1.5) which has preliminary support for macOS and Linux.  I haven't tested it yet on macOS but have it working in Linux.  FYI you may get permission denied issues compiling.

**Comment by vbauer** [comment]
Yes, it works! I only specified a path (`/usr/local/bin/stella`) to Stella (it was installed with Brew)

**Comment by chunkypixel** [comment]
Hey thats great news that it's working - thanks so much for trying this out! For Linux and macOS currently the Stella path is required to be set (hopefully you got a prompt in the output?). Once I get access to a Mac I'll see if I can include the Stella files as I'm doing with Windows - not sure about Linux though with the various flavors...

Thanks again!!

**Comment by vbauer** [comment]
> For Linux and macOS currently the Stella path is required to be set (hopefully you got a prompt in the output?)

Yes, the message was "WARNING: You must provide a path to your emulator before you can launch your game."

I've just faced with some minor problem - it seems that hot-keys don't work (probably I do something wrong):
When I press "Control + Option + N", I see an error "Code language not supported or defined." instead of action "bB: Compile batari Basic source code" (some with other shortcuts)

Thank you again! It is time to write my first game for Atari 2600 :)

**Comment by vbauer** [comment]
It also looks like that basic-file must be on the top of project folder:

1. I opened folder "pothole" with "pothole.bas" file and build project - everything is fine (pothole/pothole.bas)

2. I opened folder "atari2600" with subfolder "games" with subfolder "pothole" and tried to build pothole.bas (atari2600/games/pothole/pothole.bas):
```
Starting build of pothole.bas
/Users/vbauer/.vscode/extensions/chunkypixel.bataribasic-0.1.6/out/bin/BB.1.1d.reveng41/2600basic.sh: line 36: pothole.bas: No such file or directory
```

**Comment by chunkypixel** [comment]
vbauer,
I'll see if I can replicate that issue.
Also I need to add the macOS shortcuts - currently not configured for them as I wasn't sure if the keystrokes.

**Comment by chunkypixel** [comment]
I tried creating the same folder structure in Linux and it worked ok.  I'll need to head to my brothers at some stage to try it on a mac.  I'll open this as a new issue to track.

**Comment by vbauer** [comment]
> I tried creating the same folder structure in Linux and it worked ok

Hmm, I've just checked on Ubuntu and got the same result:

![Screenshot_2019-04-16_01-04-26](https://user-images.githubusercontent.com/578021/56154806-a27fa500-5fe3-11e9-8868-051c3754921a.png)

[FYI] This is my test workspace :) 
[atari2600.tar.gz](https://github.com/chunkypixel/batariBasic-extension/files/3081496/atari2600.tar.gz)

**Comment by chunkypixel** [comment]
Strange! Thanks for the file - will check that out. 

Sorry again for the trouble!

**Comment by chunkypixel** [comment]
Ahh... Sorry I can now see how you have explained it firstly is now very clear in the pic you provided.  It will have to do with how I'm determining certain paths.  Will get a fix out very soon!

**Comment by chunkypixel** [comment]
Build v.0.1.7
Fixed issue compiling a file which was located in a subfolder of the open workspace (thanks vbauer!)
